app/mixins/async_tasks.js

```diff
@@ -0,0 +1,61 @@
+import Ember from "ember";
+const { getOwner } = Ember;
+
+export default Ember.Mixin.create({
+  messageBox: Ember.inject.service(),
+
+  // ---- Helpers
+
+  __tasksCount: 0,
+  __loadingView: null,
+
+  __incrementTaskCount(val = 1) {
+    this.__tasksCount += val;
+    if (this.__tasksCount > 0) {
+      this.showLoadingSpinner();
+    } else {
+      this.__tasksCount = 0;
+      this.hideLoadingSpinner();
+    }
+  },
+
+  // --- Mixin api
+
+  runTask(task) {
+    this.__incrementTaskCount();
+    return task
+      .then(res => {
+        this.__incrementTaskCount(-1);
+        return res;
+      })
+      .catch(err => {
+        this.__incrementTaskCount(-1);
+        return Ember.RSVP.reject(err);
+      });
+  },
+
+  showLoadingSpinner() {
+    if (Ember.testing) {
```

Comment:
Does this mess up tests? Any way to work around it so the code does run during tests?

Reply:
Yeah, there was a reeeaaalllly dodgy failing test once.  I never did figure out another way to fix it, It had something to do with the loading spinner being removed at odd moments.

Reply:
Shame but understandable!